api/src/main/java/marquez/service/RunService.java

```diff
@@ -69,37 +69,36 @@ public void markRunAs(
     ExtendedRunRow runRow = findByRow(runId.getValue()).get();
     runStateDao.updateRunState(runId.getValue(), runState, transitionedAt);
 
-    if (runState == COMPLETED) {
+    if (runState.isDone()) {
```

Comment:
Do we want to create job versions even though the job run isn't successful? Only successful runs have output datasets.

Reply:
Hmm, this logic might make sense. A failed or aborted job will simply produce a job version without any outputs.

Reply:
Yes, there should be a version for failed job. That's how you'll be able to figure out the job has failed because a bug was introduced in the latest version for example.

Reply:
Yeah, that make sense and looks like we still maintain the logic to only notify the `RunListener` of input / output changes on job completion.